### PR TITLE
DOC: Improved docs for `jnp.nanvar` and `jnp.nanstd`

### PR DESCRIPTION
**Pull request by rajasekharporeddy**
Part of #21461

### CODE REVIEW

File: jax/_src/numpy/reductions.py [hunk 1]
```diff
@@ -1249,6 +1249,79 @@ def nanmean(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None, out
 def nanvar(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None, out: None = None,
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:1249]: Please remove the `@implements` decorator above

File: jax/_src/numpy/reductions.py [hunk 2]
```diff
@@ -1249,6 +1249,79 @@ def nanmean(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None, out
...
            ddof: int = 0, keepdims: bool = False,
            where: ArrayLike | None = None) -> Array:
+  r"""Compute the variance of array elements along a given axis, ignoring NaNs.
+
+  JAX implementation of :func:`numpy.nanvar`.
+
+  Args:
+    a: input array.
+    axis: optional, sequence of ints, default=None. Axis along which the variance
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:1258]: int or sequence of ints

File: jax/_src/numpy/reductions.py [hunk 3]
```diff
@@ -1280,6 +1353,70 @@ def nanvar(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None, out:
 def nanstd(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None, out: None = None,
            ddof: int = 0, keepdims: bool = False,
            where: ArrayLike | None = None) -> Array:
+  r"""Compute the standard deviation along a given axis, ignoring NaNs.
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:1356]: Please remove the `@implements` decorator above.

File: jax/_src/numpy/reductions.py [hunk 4]
```diff
@@ -1280,6 +1353,70 @@ def nanvar(a: ArrayLike, axis: Axis = None, dtype: DTypeLike | None = None, out:
...
+
+  JAX implementation of :func:`numpy.nanstd`.
+
+  Args:
+    a: input array.
+    axis: optional, sequence of ints, default=None. Axis along which the standard
```
Comment (jakevdp) [inline jax/_src/numpy/reductions.py:1362]: int or sequence of ints

**Review by jakevdp — CHANGES_REQUESTED** [review]
Thanks!

**Comment by rajasekharporeddy** [comment]
Modified the docstrings as suggested and squashed the commits into one.

File: jax/_src/numpy/reductions.py (hunk 4 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py]: Done

File: jax/_src/numpy/reductions.py (hunk 3 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py:1354]: Done

File: jax/_src/numpy/reductions.py (hunk 1 above)
Comment (rajasekharporeddy) [reply jax/_src/numpy/reductions.py:1248]: Done

**Review by jakevdp — APPROVED** [review]
Thanks!

**Comment by jakevdp** [comment]
Please fix the lint issues - thanks!

**Review by jakevdp — APPROVED** [review]

**Comment by jakevdp** [comment]
It looks like we need a rebase in order to merge this - thanks!

**Comment by rajasekharporeddy** [comment]
Rebased

**Pull request merged**